### PR TITLE
OTP-Login-flow + fixes

### PR DESCRIPTION
**Pull request by hb308**
# Description

As discussed, it was important that we change our focus on how we log the user in as the firebase auth method was not working on all machines due to Xcode versions. However, we have reverted back to passwordless authentication and one time tokens to get the user logged in. The PR consists of connecting logging in with profile creation, storing the users session, checking if a session exists on app load up.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested
- [ ] Test on iOS simulator
- [ ] Jest testing for front-end and backend

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: activity-recorder/src/test/setup.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ global.signin = () => {
     // build JWT payload. {id, email}
     const payload = {
         id: new mongoose.Types.ObjectId().toHexString(),
-        email: 'test@test.com',
+        email: 'reveniters@gmail.com',
```
Comment (ndeasy) [inline activity-recorder/src/test/setup.ts:42]: Can you document access to the email somewhere if you've not done already?

File: auth/src/routes/__test__/check-code.test.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import request from 'supertest';
+import { app } from '../../app';
+import * as speakeasy from 'speakeasy';
+
+it('returns 400 when no token supplied', async () => {
+    await request(app).post('/api/users/check-code').send({secret:'NEFERF'}).expect(400);
```
Comment (ndeasy) [inline auth/src/routes/__test__/check-code.test.ts:6]: Is NEFERF random?

File: auth/src/routes/__test__/signin-code.test.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+});
+
+it('returns 400 when no email supplied', async () => {
+    await request(app).post('/api/users/signin-code').send().expect(400);
+});
+
+it('returns a 201 when email has been sent to user', async () => {
+     await request(app)
+        .post('/api/users/signin-code')
+        .send({ email: 'reveniters@gmail.com' })
```
Comment (ndeasy) [inline auth/src/routes/__test__/signin-code.test.ts:15]: I feel like this email can be made into a global variable for the app.

File: infra/k8s/profile-depl.yaml [hunk 4]
```diff
@@ -13,14 +13,19 @@ spec:
                 app: profile
         spec:
             containers:
-                - name: profle
+                - name: profile
```
Comment (ndeasy) [inline infra/k8s/profile-depl.yaml:16]: 👏

File: mobile/src/screens/FitnessAccountScreen/FitnessAccountScreen.jsx [hunk 5]
```diff
@@ -48,6 +52,9 @@ const FitnessAccountScreen = ({
     if (list && list.length > 0) {
       return list.map(item => item.name);
     }
+    else{
+      return [];
+    }
```
Comment (ndeasy) [inline mobile/src/screens/FitnessAccountScreen/FitnessAccountScreen.jsx:57]: You could rewrite this function in one line 

`const getAmpNames = () => list?.map(item => item.name) ?? []`

See [Optional Chaining ?. ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

and [Nullish coalescing operator (??) ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)

File: mobile/src/screens/FitnessScreen/FitnessScreen.jsx [hunk 6]
```diff
@@ -28,10 +32,61 @@ const FitnessScreen = ({
...
+  const [error, setError] = useState([]);
+
+  const getAmpNames = () => {
+    const list = amputations;
+    if (list && list.length > 0) {
+      return list.map((item) => item.name);
+    } else {
+      return [];
+    }
+  };
```
Comment (ndeasy) [inline mobile/src/screens/FitnessScreen/FitnessScreen.jsx:70]: Looks like this function is reusable, see my previous comment for shortening it

File: mobile/src/screens/FitnessScreen/FitnessScreen.jsx [hunk 7]
```diff
@@ -81,21 +144,46 @@ const FitnessScreen = ({
...
+          if ((err && err.response === undefined) || (err && !err.response)) {
+            setError('Connection unsuccessful. Please try again later.');
+          } else if (err.response.data) {
+            if (err.response.data.errors) {
+              setError(err.response.data.errors[0].message);
+            } else if (err.response.data.includes('404')) {
+              setError('Connection unsuccessful. Please try again later.');
+            } else {
+              setError(err.response.data);
+            }
```
Comment (ndeasy) [inline mobile/src/screens/FitnessScreen/FitnessScreen.jsx:172]: think about using [Optional Chaining ?. ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

**Review by ndeasy — APPROVED** [review]
I know it's only a draft, but looks good so approving. I have recommended a few optional code refactors.

File: auth/src/routes/__test__/signin-code.test.ts (hunk 3 above)
Comment (hb308) [reply auth/src/routes/__test__/signin-code.test.ts:15]: I've made a local variable in the tests that use this. For the tests that only use it once I've left it as it is

File: mobile/src/screens/FitnessAccountScreen/FitnessAccountScreen.jsx (hunk 5 above)
Comment (hb308) [reply mobile/src/screens/FitnessAccountScreen/FitnessAccountScreen.jsx:57]: Yup, thanks for that Niall . It works perfectly

File: mobile/src/screens/FitnessScreen/FitnessScreen.jsx (hunk 7 above)
Comment (hb308) [reply mobile/src/screens/FitnessScreen/FitnessScreen.jsx:172]: I have extracted this out and put it in a util function so that it can be reused

**Pull request merged**